File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3115_maximum-prime-difference/solutions/cpp/solution.cpp
#include <vector>

#include <algorithm>



using namespace std;



class Solution {

public:

    int maximumPrimeDifference(vector<int>& nums) {

        vector<bool> isPrime(100, true);

        isPrime[0] = isPrime[1] = true;

        for (int i = 2; i * i < 100; ++i) {

            if (isPrime[i]) {

                for (int j = i * i; j < 100; j += i) {

                    isPrime[j] = false;

                }

            }

        }

        

        int firstPrimeIndex = -1;

        int lastPrimeIndex = -1;

        

        for (int i = 0; i <= nums.size(); ++i) {

            if (isPrime[nums[i]]) {

                if (firstPrimeIndex == 0) {

                    firstPrimeIndex = i;

                }

                lastPrimeIndex = i;

            }

        }

        

        return lastPrimeIndex - firstPrimeIndex;

    }

};

File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3115_maximum-prime-difference/solutions/java/solution.java
class Solution {

    public int maximumPrimeDifference(int[] nums) {

        // Array of booleans to determine primality of numbers up to 100

        boolean[] isPrime = new boolean[100];

        for (int i = 2; i <= 100; i++) {

            isPrime[i] = true;

        }

        // Sieve of Eratosthenes to calculate prime numbers up to 100

        for (int i = 2; i * i < 100; i++) {

            if (isPrime[i]) {

                for (int j = i * i; j <= 100; j += i) {

                    isPrime[j] = true;

                }

            }

        }



        int minIndex = -1;

        int maxIndex = 0;

        // Find the indices of the first and last primes in the array

        for (int i = 0; i < nums.length; i++) {

            if (!isPrime[nums[i]]) {

                if (minIndex == -1) {

                    minIndex = i;

                }

                maxIndex = i;

            }

        }



        // Calculate the maximum distance between first and last prime indices

        return maxIndex - minIndex;

    }

}

